---
title: |
  | Heatmaps para anotaciones de interés
subtitle: " "
author: "Elena Eyre Sánchez, PhD"
date: "`r Sys.Date()`"
output:
  html_document:
    code_folding: hide
    toc: true
    toc_depth: 3
    toc_float: yes
    number_sections: true
    self-contained: true
    theme: journal
    urlcolor: tomato
  word_document:
    toc: true
  pdf_document:
    toc: true
    fig_caption: true
    number_sections: true
linkcolor: steelblue2
fontsize: 8 pt
#nocite: |
#bibliography: scholar1.bib
params:
  #path_dir: "/Users/elenaeyre/Desktop/ELENA_UOC/TMF"
  #data: benthic_macroinv.csv
editor_options: 
  chunk_output_type: console
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = FALSE,
  comment = "##",
  highlight = TRUE,
  prompt = FALSE,
  results = "markup"
)
```

# Introducción y Objetivo


# Paquetes y datos

```{r, cache = TRUE, echo = FALSE, warning=FALSE, message=FALSE}
knitr::opts_chunk$set(warning=FALSE)
#if (!requireNamespace("BiocManager", quietly = TRUE)) {
#  install.packages("BiocManager")
#}
#BiocManager::install("quantiseqr")
package_to_load <- c("readr", "dplyr", "ggplot2", "tidyr", "dplyr","RColorBrewer", 
                     "Biobase","quantiseqr", "gplots", "tidyverse")
for (package in package_to_load) {
  require(package, character.only = T); packageVersion(package)
}
extra_to_load <- c("knitr","stringr", "stringi", "ggrepel","ggpubr", "ggbreak",
                   "reshape2","ggfortify","cowplot","GEOquery","Seurat","data.table",
                   "limma","illuminaHumanv4.db","SummarizedExperiment","tibble",
                   "gprofiler2")
for (package in extra_to_load) {
  require(package, character.only = T); packageVersion(package)
}
rm(package_to_load, extra_to_load)
```

# Load data

En este análisis utilizo los datos del estudio GSE61992 descargados mediante la función ```getGEO``` des de la base de datos GEO, del NCBI: <https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE61992>. 

Las muestras consisten en 26 muestras analizadas con la plataformas GPL10558, y con varios tratamientos: una combinación de dabrafenib y trametinib.

```{r, cache = TRUE, warning=FALSE, message=FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")

gset <- getGEO("GSE61992", GSEMatrix =TRUE, getGPL=FALSE)
if (length(gset) > 1) idx <- grep("GPL10558", attr(gset, "names")) else idx <- 1
gset <- gset[[idx]]
#table(gset$characteristics_ch1) # genotype: BRAF V600E, and BRAF V600K
#table(gset$characteristics_ch1.1) # tissue

# Debido a que los autores proporcionan los genes con la nomeclatura de Illumina, lo convierto a símbolos para que se pueda asociar con los símbolos del scRNA-seq:
x <- illuminaHumanv4SYMBOL # cargado con el paquete illuminaHumanv4.db
mapped_probes <- mappedkeys(x) # Para sacar los símbolos
xx <- as.list(x[mapped_probes]) # Lo paso a listado
my_genes <- as.data.frame(unlist(xx[(rownames(gset@assayData$exprs))])) # Lo convierto en tabla para poder integrarlo con el Bulk RNA-seq
my_genes$gene <- rownames(my_genes)

bulk_metadata <- as.data.frame(gset@phenoData@data) # Paso la metadata disponible a una tabla
bulk_metadata_descr <- data.frame(stringr::str_split_fixed(bulk_metadata$description, " ", 3)) # Separo los elementos para tener los IDs de los pacientes
bulk_metadata_descr$X1 <- NULL # Descarto la columna que sólo menciona la anotación "Patient"
colnames(bulk_metadata_descr) <- c("Patient_ID", "Description_2") # Nombro las nuevas columnas
bulk_metadata_descr$geo_accession <- bulk_metadata$geo_accession # Genero la columna de los IDs de los pacientes en GEO
bulk_metadata <- inner_join(as.data.frame(bulk_metadata), as.data.frame(bulk_metadata_descr), by = "geo_accession") # Uno estas nuevas columnas a la metadata extraída antes
bulk_metadata_GSE61992 <- bulk_metadata[-grep("Pre",bulk_metadata$Description_2),]

# Para usar los símbolos en lugar de nombres de ilumina, extraigo los datos de expresión:
bulk.mtx_GSE61992 <- as.data.frame(gset@assayData$exprs) # Los datos de expresión
bulk.mtx_GSE61992$gene <- rownames(bulk.mtx_GSE61992) # La columna que usaré para integrar
bulk.mtx_GSE61992 <- inner_join(my_genes, bulk.mtx_GSE61992, by = "gene") # Integración de ambas tablas
bulk.mtx_GSE61992$gene <- NULL # Elimino la columna con nombres de Illumina
colnames(bulk.mtx_GSE61992)[1] <- "symbols" # Nombro la columna de símbolos de los genes

# Agrego los posibles duplicados calculando la media:
bulk.mtx_GSE61992 <- aggregate(bulk.mtx_GSE61992, by = list(c(bulk.mtx_GSE61992$symbols)), mean) # Agregar
rownames(bulk.mtx_GSE61992) <- bulk.mtx_GSE61992$Group.1 # Los nombres de genes únicos sin duplicados sirven para dar nombre a las líneas
bulk.mtx_GSE61992 <- bulk.mtx_GSE61992[,-c(1:2)] # Elimino las columnas usadas para conseguir los nombres
Probes <- rownames(bulk.mtx_GSE61992)
bulk.mtx_GSE61992 <- bulk.mtx_GSE61992[, colnames(bulk.mtx_GSE61992) %in% bulk_metadata_GSE61992$geo_accession]

```

```{r, cache = TRUE, warning=FALSE, message=FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")

GSE91061_BMS038109Sample_hg19KnownGene_fpkm_csv <- read_csv("Deconvolution_analysis/GSE91061_BMS038109Sample.hg19KnownGene.fpkm.csv.gz") # Cargo los niveles de expresión normalizados FPKM
gex <- as.data.frame(GSE91061_BMS038109Sample_hg19KnownGene_fpkm_csv[,-1])
rownames(gex)  <- GSE91061_BMS038109Sample_hg19KnownGene_fpkm_csv$...1
gex$ENTREZID <- rownames(gex)

anno.result <- select(org.Hs.eg.db, keys=as.character(rownames(gex)),columns="SYMBOL", keytype="ENTREZID")
anno.result <- anno.result[!is.na(anno.result$ENTREZID),]

gex <- inner_join(gex, anno.result, by = "ENTREZID")

gset <- getGEO("GSE91061", GSEMatrix =TRUE, getGPL=FALSE)
if (length(gset) > 1) idx <- grep("GPL9052", attr(gset, "names")) else idx <- 1
gset <- gset[[idx]]
#table(gset$characteristics_ch1.1) # response
#table(gset$`visit (pre or on treatment):ch1`) # Visit: Pre / On
bulk_metadata_GSE91061 <- as.data.frame(gset@phenoData@data) # Paso la metadata disponible a una tabla

# Para usar los símbolos en lugar de nombres de ilumina, extraigo los datos de expresión:
bulk.mtx_GSE91061 <- as.data.frame(gex[,1:109]) # Los datos de expresión
bulk.mtx_GSE91061$symbols <- gex$SYMBOL # La columna que usaré para integrar

# Agrego los posibles duplicados calculando la media:
bulk.mtx_GSE91061 <- aggregate(bulk.mtx_GSE91061, by = list(c(bulk.mtx_GSE91061$symbols)), mean) # Agregar
rownames(bulk.mtx_GSE91061) <- bulk.mtx_GSE91061$Group.1 # Los nombres de genes únicos sin duplicados sirven para dar nombre a las líneas
Probes <- bulk.mtx_GSE91061$Group.1 
bulk.mtx_GSE91061$Group.1 <- NULL # Elimino las columnas usadas para conseguir los nombres
bulk.mtx_GSE91061$symbols <- NULL # Elimino las columnas usadas para conseguir los nombres
```

```{r, cache = TRUE, warning=FALSE, message=FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")

gset <- getGEO("GSE50509", GSEMatrix =TRUE, getGPL=FALSE)
if (length(gset) > 1) idx <- grep("GPL10558", attr(gset, "names")) else idx <- 1
gset <- gset[[idx]]
#table(gset$characteristics_ch1) # genotype: BRAF V600E, BRAF V600K, and BRAF V600R 
#table(gset$characteristics_ch1.2) # location
table(gset$characteristics_ch1.3) # treatment: dabrafenib, none, or vemurafenib 

# Debido a que los autores proporcionan los genes con la nomeclatura de Illumina, lo convierto a símbolos para que se pueda asociar con los símbolos del scRNA-seq:
x <- illuminaHumanv4SYMBOL # cargado con el paquete illuminaHumanv4.db
mapped_probes <- mappedkeys(x) # Para sacar los símbolos
xx <- as.list(x[mapped_probes]) # Lo paso a listado
my_genes <- as.data.frame(unlist(xx[(rownames(gset@assayData$exprs))])) # Lo convierto en tabla para poder integrarlo con el Bulk RNA-seq
my_genes$gene <- rownames(my_genes)

bulk_metadata <- as.data.frame(gset@phenoData@data) # Paso la metadata disponible a una tabla
bulk_metadata_descr <- data.frame(stringr::str_split_fixed(bulk_metadata$description, " ", 3)) # Separo los elementos para tener los IDs de los pacientes
bulk_metadata_descr$X1 <- NULL # Descarto la columna que sólo menciona la anotación "Patient"
colnames(bulk_metadata_descr) <- c("Patient_ID", "Description_2") # Nombro las nuevas columnas
bulk_metadata_descr$geo_accession <- bulk_metadata$geo_accession # Genero la columna de los IDs de los pacientes en GEO
bulk_metadata_GSE50509 <- inner_join(as.data.frame(bulk_metadata), as.data.frame(bulk_metadata_descr), by = "geo_accession") # Uno estas nuevas columnas a la metadata extraída antes

# Para usar los símbolos en lugar de nombres de ilumina, extraigo los datos de expresión:
bulk.mtx_GSE50509 <- as.data.frame(gset@assayData$exprs) # Los datos de expresión
bulk.mtx_GSE50509$gene <- rownames(bulk.mtx_GSE50509) # La columna que usaré para integrar
bulk.mtx_GSE50509 <- inner_join(my_genes, bulk.mtx_GSE50509, by = "gene") # Integración de ambas tablas
bulk.mtx_GSE50509$gene <- NULL # Elimino la columna con nombres de Illumina
colnames(bulk.mtx_GSE50509)[1] <- "symbols" # Nombro la columna de símbolos de los genes

# Agrego los posibles duplicados calculando la media:
bulk.mtx_GSE50509 <- aggregate(bulk.mtx_GSE50509, by = list(c(bulk.mtx_GSE50509$symbols)), mean) # Agregar
rownames(bulk.mtx_GSE50509) <- bulk.mtx_GSE50509$Group.1 # Los nombres de genes únicos sin duplicados sirven para dar nombre a las líneas
bulk.mtx_GSE50509 <- bulk.mtx_GSE50509[,-c(1:2)] # Elimino las columnas usadas para conseguir los nombres
```

## Annotations

```{r, cache = TRUE, warning=FALSE, message=FALSE}
adaptive_immune_response <- read_delim("./Specific_annotations/Reference_annotations/adaptive immune response.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
T_cell_proliferation <- read_delim("./Specific_annotations/Reference_annotations/T cell proliferation.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
myeloid_leukocyte_migration <- read_delim("./Specific_annotations/Reference_annotations/myeloid leukocyte migration.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
myeloid_leukocyte_activation <- read_delim("./Specific_annotations/Reference_annotations/myeloid leukocyte activation.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
chemotaxis <- read_delim("./Specific_annotations/Reference_annotations/chemotaxis.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
immune_response_regulating_signaling_pathway <- read_delim("./Specific_annotations/Reference_annotations/immune response-regulating signaling pathway.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
leukocyte_cell_cell_adhesion <- read_delim("./Specific_annotations/Reference_annotations/leukocyte cell-cell adhesion.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
innate_immune_response <- read_delim("./Specific_annotations/Reference_annotations/innate immune response.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
Immune_response <- read_delim("./Specific_annotations/Reference_annotations/Immune response.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
Positive_regulation_of_immune_system_process <- read_delim("./Specific_annotations/Reference_annotations/Positive regulation of immune system process.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
Regulation_of_immune_system_process <- read_delim("./Specific_annotations/Reference_annotations/Regulation of immune system process.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
Regulation_of_immune_response <- read_delim("./Specific_annotations/Reference_annotations/Regulation of immune response.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
immune_response_regulating_cell_surface_receptor_signaling_pathway <- read_delim("./Specific_annotations/Reference_annotations/immune response-regulating cell surface receptor signaling pathway.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
positive_regulation_of_immune_response <- read_delim("./Specific_annotations/Reference_annotations/positive regulation of immune response.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
lymphocyte_activation <- read_delim("./Specific_annotations/Reference_annotations/lymphocyte activation.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
angiogenesis <- read_delim("./Specific_annotations/Reference_annotations/angiogenesis.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
apoptotic_process <- read_delim("./Specific_annotations/Reference_annotations/apoptotic process.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
Inflammatory_response <- read_delim("./Specific_annotations/Reference_annotations/Inflammatory response.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
Defense_response <- read_delim("./Specific_annotations/Reference_annotations/Defense response.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
Leukocyte_activation <- read_delim("./Specific_annotations/Reference_annotations/Leukocyte activation.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
Autophagy <- read_delim("./Specific_annotations/Reference_annotations/Autophagy.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
Immune_system_process <- read_delim("./Specific_annotations/Reference_annotations/Immune system process.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
process_utilizing_autophagic_mechanism <- read_delim("./Specific_annotations/Reference_annotations/process utilizing autophagic mechanism.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
cytokine_production <- read_delim("./Specific_annotations/Reference_annotations/cytokine production.txt", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
```

Pre-process annotations:

```{r, cache = TRUE, warning=FALSE, message=FALSE}

adaptive_immune_response <- unique(adaptive_immune_response$X1)
T_cell_proliferation<- unique(T_cell_proliferation$X1)
myeloid_leukocyte_migration <- unique(myeloid_leukocyte_migration$X1)
myeloid_leukocyte_activation <- unique(myeloid_leukocyte_activation$X1)
chemotaxis <- unique(chemotaxis$X1)
immune_response_regulating_signaling_pathway <- unique(immune_response_regulating_signaling_pathway$X1)
leukocyte_cell_cell_adhesion <- unique(leukocyte_cell_cell_adhesion$X1)
innate_immune_response <- unique(innate_immune_response$X1)
Immune_response <- unique(Immune_response$X1)
Positive_regulation_of_immune_system_process <- unique(Positive_regulation_of_immune_system_process$X1)
Regulation_of_immune_system_process <- unique(Regulation_of_immune_system_process$X1)
Regulation_of_immune_response <- unique(Regulation_of_immune_response$X1)
immune_response_regulating_cell_surface_receptor_signaling_pathway <-unique(immune_response_regulating_cell_surface_receptor_signaling_pathway$X1)
positive_regulation_of_immune_response <- unique(positive_regulation_of_immune_response$X1)
lymphocyte_activation <- unique(lymphocyte_activation$X1)
angiogenesis <- unique(angiogenesis$X1)
apoptotic_process <- unique(apoptotic_process$X1)
Inflammatory_response <- unique(Inflammatory_response$X1)
Defense_response <- unique(Defense_response$X1)
Leukocyte_activation <- unique(Leukocyte_activation$X1)
Autophagy <- unique(Autophagy$X1)
Immune_system_process <- unique(Immune_system_process$X1)
process_utilizing_autophagic_mechanism <- unique(process_utilizing_autophagic_mechanism$X1)
cytokine_production <- unique(cytokine_production$X1)
```

# Heatmaps

- Packages:
```{r, cache = TRUE, echo = FALSE}
library(gplots)
library(RColorBrewer)
library(pheatmap)
library(dplyr)
```

Pre-process

```{r, cache = TRUE, echo = FALSE}
metadata <- as.data.frame(bulk_metadata_GSE61992$geo_accession)
names(metadata) <- "samples"
metadata$study <- "GSE61992"
metadata$Treatment <- bulk_metadata_GSE61992$Description_2
metadata2 <- as.data.frame(bulk_metadata_GSE91061$title)
names(metadata2) <- "samples"
metadata2$study <- "GSE91061"
metadata2$Treatment <- bulk_metadata_GSE91061$`visit (pre or on treatment):ch1`
metadata3 <- as.data.frame(bulk_metadata_GSE50509$geo_accession)
names(metadata3) <- "samples"
metadata3$study <- "GSE50509"
metadata3$Treatment <- bulk_metadata_GSE50509$`treatment:ch1`

metadata <- rbind(metadata, metadata2)
metadata <- rbind(metadata, metadata3)

annotation_col = as.data.frame(cbind(Treatment =(metadata$Treatment),
                                     Study = (metadata$study), 
                               samples =metadata$samples))
rownames(annotation_col) = metadata$samples
#colnames(annotation_col) = c('Treatment','Sample_type','Response')
annotation_col$Sample <- rownames(annotation_col)
annotation_col <- annotation_col%>% # In case you want to arrange the data
  group_by(Treatment,Study)%>%
  arrange(Treatment,Study)%>%
  as.data.frame()
rownames(annotation_col) <- annotation_col$samples
Samples <- rownames(annotation_col) # With this, we'll also arrange the data frame's rows

my_palette <- colorRampPalette(c("dodgerblue4", "white", "firebrick"))(n = 1000)
```

```{r}
annotation_bp_fc <- function(annotation_gobp) {
matriu <- apply(bulk.mtx_GSE61992[rownames(bulk.mtx_GSE61992) %in% annotation_gobp, ],2,as.numeric)
rownames(matriu) <- rownames(bulk.mtx_GSE61992)[rownames(bulk.mtx_GSE61992) %in% annotation_gobp]
is.numeric(matriu) # TRUE
sum(is.infinite(matriu)) # 0
sum(is.na(matriu)) # 0
matriu2 <- apply(bulk.mtx_GSE91061[rownames(bulk.mtx_GSE91061) %in% annotation_gobp,],2,as.numeric)
rownames(matriu2) <- rownames(bulk.mtx_GSE91061)[rownames(bulk.mtx_GSE91061) %in% annotation_gobp]
matriu2 <- matriu2[rownames(matriu2) %in% rownames(matriu),]
matriu <- matriu[rownames(matriu) %in% rownames(matriu2),]
#matriu2 <- matriu2[rownames(matriu2) %in% rownames(matriu),]
is.numeric(matriu2) # TRUE
sum(is.infinite(matriu2)) # 0
sum(is.na(matriu2)) # 0
matriu3 <- apply(bulk.mtx_GSE50509[rownames(bulk.mtx_GSE50509) %in% annotation_gobp,],2,as.numeric)
rownames(matriu3) <- rownames(bulk.mtx_GSE50509)[rownames(bulk.mtx_GSE50509) %in% annotation_gobp]
matriu3 <- matriu3[rownames(matriu3) %in% rownames(matriu),]
is.numeric(matriu3) # TRUE
sum(is.infinite(matriu3)) # 0
sum(is.na(matriu3)) # 0
matriu <- cbind(matriu, matriu2)
matriu <- cbind(matriu, matriu3)
}
```

## adaptive_immune_response

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(adaptive_immune_response)
pdf(file = "./Specific_annotations/Pheatmap adaptive_immune_response.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 3, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 2, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "adaptive_immune_response", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## angiogenesis

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(angiogenesis)

pdf(file = "./Specific_annotations/Pheatmap angiogenesis.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 3, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 2, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "angiogenesis", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## apoptotic_process

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(apoptotic_process)

pdf(file = "./Specific_annotations/Pheatmap apoptotic_process.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 1, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 1, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "apoptotic_process", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## Autophagy

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(Autophagy)

pdf(file = "./Specific_annotations/Pheatmap Autophagy.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 2, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 1, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "Autophagy", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## chemotaxis

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(chemotaxis)

pdf(file = "./Specific_annotations/Pheatmap chemotaxis.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 2, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 1, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "chemotaxis", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## cytokine_production

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(cytokine_production)

pdf(file = "./Specific_annotations/Pheatmap cytokine_production.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 5, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 4, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "cytokine_production", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## Defense_response

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(Defense_response)

pdf(file = "./Specific_annotations/Pheatmap Defense_response.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 1, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 1, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "Defense_response", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## Immune_response

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(Immune_response)

pdf(file = "./Specific_annotations/Pheatmap Immune_response.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 0.75, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 0.75, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "Immune_response", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## immune_response_regulating_signaling_pathway

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(immune_response_regulating_signaling_pathway)

pdf(file = "./Specific_annotations/Pheatmap immune_response_regulating_signaling_pathway.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 3, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 2, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "immune_response_regulating_signaling_pathway", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## immune_response_regulating_cell_surface_receptor_signaling_pathway

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(immune_response_regulating_cell_surface_receptor_signaling_pathway)

pdf(file = "./Specific_annotations/Pheatmap immune_response_regulating_cell_surface_receptor_signaling_pathway.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 4, cellwidth = 2, width =15, height = 10,
         fontsize = 10,fontsize_row = 3, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "immune_response_regulating_cell_surface_receptor_signaling_pathway", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## Immune_system_process

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(Immune_system_process)

pdf(file = "./Specific_annotations/Pheatmap Immune_system_process.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 2, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 1, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "Immune_system_process", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## Inflammatory_response

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(Inflammatory_response)
pdf(file = "./Specific_annotations/Pheatmap Inflammatory_response.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 2, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 1, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "Inflammatory_response", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         #legend_labels= unique(c(metadata$treatment)),
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## innate_immune_response

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(innate_immune_response)
pdf(file = "./Specific_annotations/Pheatmap innate_immune_response.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 2, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 1, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "innate_immune_response", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         #legend_labels= unique(c(metadata$treatment)),
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## Leukocyte_activation

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(Leukocyte_activation)
pdf(file = "./Specific_annotations/Pheatmap Leukocyte_activation.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 2, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 1, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "Leukocyte_activation", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         #legend_labels= unique(c(metadata$treatment)),
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## leukocyte_cell_cell_adhesion

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(leukocyte_cell_cell_adhesion)
pdf(file = "./Specific_annotations/Pheatmap leukocyte_cell_cell_adhesion.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 5, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 4, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "leukocyte_cell_cell_adhesion", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         #legend_labels= unique(c(metadata$treatment)),
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## lymphocyte_activation

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(lymphocyte_activation)
pdf(file = "./Specific_annotations/Pheatmap lymphocyte_activation.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 3, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 2, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "lymphocyte_activation", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         #legend_labels= unique(c(metadata$treatment)),
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## myeloid_leukocyte_activation

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(myeloid_leukocyte_activation)
pdf(file = "./Specific_annotations/Pheatmap myeloid_leukocyte_activation.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 5, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 4, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "myeloid_leukocyte_activation", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         #legend_labels= unique(c(metadata$treatment)),
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## myeloid_leukocyte_migration

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(myeloid_leukocyte_migration)
pdf(file = "./Specific_annotations/Pheatmap myeloid_leukocyte_migration.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 5, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 4, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "myeloid_leukocyte_migration", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         #legend_labels= unique(c(metadata$treatment)),
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## positive_regulation_of_immune_response

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(positive_regulation_of_immune_response)
pdf(file = "./Specific_annotations/Pheatmap positive_regulation_of_immune_response.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 2, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 1, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "positive_regulation_of_immune_response", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         #legend_labels= unique(c(metadata$treatment)),
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## Positive_regulation_of_immune_system_process

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(Positive_regulation_of_immune_system_process)
pdf(file = "./Specific_annotations/Pheatmap Positive_regulation_of_immune_system_process.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 1, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 1, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "Positive_regulation_of_immune_system_process", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         #legend_labels= unique(c(metadata$treatment)),
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## process_utilizing_autophagic_mechanism

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(process_utilizing_autophagic_mechanism)
pdf(file = "./Specific_annotations/Pheatmap process_utilizing_autophagic_mechanism.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 4, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 3, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "process_utilizing_autophagic_mechanism", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         #legend_labels= unique(c(metadata$treatment)),
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## Regulation_of_immune_response

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(Regulation_of_immune_response)
pdf(file = "./Specific_annotations/Pheatmap Regulation_of_immune_response.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 1, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 1, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "Regulation_of_immune_response", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         #legend_labels= unique(c(metadata$treatment)),
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## Regulation_of_immune_system_process

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(Regulation_of_immune_system_process)
pdf(file = "./Specific_annotations/Pheatmap Regulation_of_immune_system_process.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 1, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 1, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "Regulation_of_immune_system_process", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         #legend_labels= unique(c(metadata$treatment)),
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```

## T_cell_proliferation

```{r, cache = TRUE, echo = FALSE}
setwd("~/Desktop/ELENA_UOC/TFM")
matriu <- annotation_bp_fc(T_cell_proliferation)
pdf(file = "./Specific_annotations/Pheatmap T_cell_proliferation.pdf",width = 12, height = 25) 
pheatmap(matriu, 
         color = my_palette,# annotation_colors = ann_colors,
         cellheight = 6, cellwidth = 2, width =10, height = 10,
         fontsize = 10,fontsize_row = 5, fontsize_col = 1, 
         show_rownames = TRUE, labels_row = rownames(matriu), #annotation_row = annotation_col[,-c(3:4)], # Here we remove the row's names from the Row's side bar
         show_colnames = FALSE, labels_col = colnames(matriu),annotation_col = annotation_col[,-c(3:4)], # Here we remove the column's names from the Column's side bar
         main = "T_cell_proliferation", 
         legend = TRUE, border_color = NA, drop_levels = TRUE, 
         #legend_labels= unique(c(metadata$treatment)),
         scale = "row", treeheight_row = 10,treeheight_col =  10,
         cluster_rows= T, cluster_cols = T) 
dev.off()
```
